### PR TITLE
feat(profiling) move upscaling to libdatadog

### PR DESCRIPTION
**Pull request by realFlowControl**
### Description

This will remove upscaling for `alloc-size` and `alloc-samples` profile types from the library and make use of the upscaling capabilities in `libdatadog`.
Doing this solves the rounding issues with the current implementation due to float -> integer conversion, therefore allocation profiles will be more correct and it offloads upscaling from the main thread to the serialization step.
Also bumps `libdatadog` to `2.1.0`

### Readiness checklist
- [x] (only for Members) Changelog has been added to the release document.
- [ ] Tests added for this feature/bug.

### Reviewer checklist
- [x] Appropriate labels assigned.
- [x] Milestone is set.
- [ ] ~Changelog has been added to the release document. For community contributors the reviewer is in charge of this task.~

### CODE REVIEW

File: profiling/src/profiling/mod.rs [hunk 1]
```diff
@@ -230,8 +230,22 @@ impl TimeCollector {
...
+        let mut offset = 0;
+        sample_types.iter().for_each(|sample_type| {
+            if sample_type.r#type == "alloc-samples" {
+                alloc_samples_offset = Some(offset);
+            }
+            if sample_type.r#type == "alloc-size" {
+                alloc_size_offset = Some(offset);
+            }
+            offset = offset + 1;
+        });
```
[the remark this answers is not in the document]
Comment (morrisonlevi) [reply profiling/src/profiling/mod.rs:245]: There's gotta be a better method; this is Rust, after all! I'll take a peek.

**Review by morrisonlevi — COMMENTED** [review]
Do we have an understanding of how this changes performance?

**Comment by realFlowControl** [comment]
Did a benchmark on CLI using a script that does a lot of allocations (therefor a lot of samples in allocation profiling)

```bash
$ hyperfine --runs 20 --warmup 2 "php -d extension=datadog-profiling-old.dylib test.php" "php -d extension=datadog-profiling.dylib test.php" 
Benchmark 1: php -d extension=datadog-profiling-old.dylib test.php
  Time (mean ± σ):     420.1 ms ±   5.1 ms    [User: 461.9 ms, System: 29.1 ms]
  Range (min … max):   416.6 ms … 436.3 ms    20 runs

Benchmark 2: php -d extension=datadog-profiling.dylib test.php
  Time (mean ± σ):     416.2 ms ±   2.4 ms    [User: 460.5 ms, System: 26.3 ms]
  Range (min … max):   414.7 ms … 425.5 ms    20 runs

Summary
  'php -d extension=datadog-profiling.dylib test.php' ran
    1.01 ± 0.01 times faster than 'php -d extension=datadog-profiling-old.dylib test.php'
```

The `-old`-version is with upscaling in the library, the other version is with upscaling in libdatadog. This is tuned to write the pprof file to disk and not send it to an agent, but this involves also the profiler startup and shutdown time, which includes also the serialisation (and therefor upscaling) in both cases.
I'll try and make another check with siege against a webserver to see those numbers as well.

**Comment by realFlowControl** [comment]
The old version with upscaling in the library:
```bash
Transactions:		         58 hits
Availability:		     100.00 %
Elapsed time:		      10.16 secs
Data transferred:	       1.50 MB
Response time:		       0.17 secs
Transaction rate:	       5.71 trans/sec
Throughput:		       0.15 MB/sec
Concurrency:		       0.99
Successful transactions:          58
Failed transactions:	          0
Longest transaction:	       0.31
Shortest transaction:	       0.14
```

The new version with upscaling in `libdatadog`
```bash
Transactions:		         63 hits
Availability:		     100.00 %
Elapsed time:		      10.87 secs
Data transferred:	       1.63 MB
Response time:		       0.17 secs
Transaction rate:	       5.80 trans/sec
Throughput:		       0.15 MB/sec
Concurrency:		       0.99
Successful transactions:          63
Failed transactions:	          0
Longest transaction:	       0.30
Shortest transaction:	       0.14
```

The difference in the `siege` benchmarks is pretty slim. 

_Note_ in the benchmarks above that the "new" version was running `0.71` seconds longer, which in worst case allows for 2 additional requests, in best case for 5 additional requests. I ran these tests multiple times and they where all close, but "leaning" towards the "new" version performing slightly better.

**Pull request merged**